src/shared/queue_linked_op.c

```diff
@@ -94,6 +97,10 @@ void * linked_queue_pop(w_linked_queue_t * queue) {
         data = queue->first->data;
         linked_queue_pop_node(queue);
     }
+
+    if (queue->pop_callback) {
```

Comment:
Just wondering whether we should call the callback if `data == NULL`. If so, maybe we should clarify in the oxygen of the header that the callback could receive a NULL pointer and that is the responsibility of the callback the sanity check.

Reply:
I think that `data` may be **NULL** in some cases, for example, when the client tries to get a value from the queue and it's empty. Maybe in that moment a callback can execute an useful action.

So I'll add the warning in the doxygen.